presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/optimization/TestJdbcJoinPushdown.java

```diff
@@ -0,0 +1,280 @@
+    {
+        return type.equals(BOOLEAN) ? booleanJdbcColumnHandle(name) : integerJdbcColumnHandle(name);
+    }
+
+    private static JdbcColumnHandle integerJdbcColumnHandle(String name)
+    {
+        return new JdbcColumnHandle(CONNECTOR_ID, name, new JdbcTypeHandle(Types.BIGINT, "integer", 10, 0), BIGINT, false, Optional.empty(), Optional.empty());
+    }
+
+    private static JdbcColumnHandle booleanJdbcColumnHandle(String name)
+    {
+        return new JdbcColumnHandle(CONNECTOR_ID, name, new JdbcTypeHandle(Types.BOOLEAN, "boolean", 1, 0), BOOLEAN, false, Optional.empty(), Optional.empty());
+    }
+
+    private TableScanNode combinedJdbcTableScan(Type type, String... columnNames)
+    {
+        Set<ConnectorTableHandle> tableHandles = new LinkedHashSet<>();
+        final String aliasPrefix = "T";
+        for (int i = 1; i <= 3; i++) {
+            String table = "test_table" + i;
+            String schema = "test_schema" + i;
+            JdbcTableHandle tableHandle = new JdbcTableHandle(CONNECTOR_ID, new SchemaTableName(schema, table), CATALOG_NAME, schema, table, Collections.emptyList(), Optional.of(aliasPrefix + i));
+            tableHandles.add(tableHandle);
+        }
+        JdbcTableHandle jdbcTableHandle = (JdbcTableHandle) tableHandles.iterator().next();
+        Set<JoinTableInfo> joinTableInfos = new LinkedHashSet<>();
+        List<VariableReferenceExpression> outputVariables = Arrays.stream(columnNames).map(column -> newVariable(column, type)).collect(toImmutableList());
+        Map<VariableReferenceExpression, ColumnHandle> assignments = Arrays.stream(columnNames)
+                .map(column -> newVariable(column, type)).collect(toMap(identity(), entry -> getColumnHandleForVariable(entry.getName(), type)));
+        tableHandles.forEach(tableHandle -> {
+            joinTableInfos.add(new JoinTableInfo(tableHandle, assignments, outputVariables));
+        });
+        ConnectorTableHandle connectorTableHandle = new JoinTableSet(ImmutableSet.copyOf(joinTableInfos));
+
+        Optional<JdbcExpression> additionalExpression = Optional.of(new JdbcExpression("(('c1' + 'c2') - 'c2')"));
+        JdbcTableLayoutHandle jdbcTableLayoutHandle = new JdbcTableLayoutHandle(TEST_SESSION.getSqlFunctionProperties(), jdbcTableHandle, TupleDomain.none(), additionalExpression);
+        TableHandle tableHandle = new TableHandle(new ConnectorId(CATALOG_NAME), connectorTableHandle, new ConnectorTransactionHandle() {}, Optional.of(jdbcTableLayoutHandle));
+
+        return PLAN_BUILDER.tableScan(
+                tableHandle,
+                outputVariables,
+                assignments);
+    }
+
+    private TableScanNode jdbcTableScan(String schema, String table, Type type, boolean isJoinPushdown, String... columnNames)
+    {
+        JdbcTableHandle jdbcTableHandle = null;
+        ConnectorTableHandle connectorTableHandle = null;
+        if (isJoinPushdown) {
```

Comment:
`isJoinPushdown` is always false in current usage. You can delete the parameter as well as the if-true block amd just retain the else part